src/main/java/edu/iis/powp/factory/Pattern.java

```diff
@@ -1,5 +1,8 @@
 package edu.iis.powp.factory;
 
+import edu.iis.client.plottermagic.IPlotter;
+
 public interface Pattern {
-    void draw();
+
+    void draw(IPlotter plotter);
```

Comment:
It looks like a duplicate of PlotterCommand interface. It should be command factory.